monai/data/dataset_summary.py

```diff
@@ -92,8 +100,8 @@ def get_target_spacing(self, spacing_key: str = "pixdim", anisotropic_threshold:
             self.collect_meta_data()
         if spacing_key not in self.all_meta_data[0]:
             raise ValueError("The provided spacing_key is not in self.all_meta_data.")
-
-        all_spacings = torch.cat([data[spacing_key][:, 1:4] for data in self.all_meta_data], dim=0).numpy()
+        all_spacings = concatenate(to_cat=[data[spacing_key][:, 1:4] for data in self.all_meta_data], axis=0)
```

Comment:
Thanks, I think this class could be enhanced with the recent 'Cumulative' API, it will ensure torch tensors.

Reply:
Ah, yes, let me try to update it with the `append` API of `Cumulative`:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/metrics/metric.py#L215

Thanks for your reminder.

Reply:
Hi @wyli ,

I spent much time today to try to use `Cumulative`, but seems it's not very suitable for this DatasetSummary.
Maybe I misunderstood your suggestion, I tried to use `Cumulative` for the `all_spacings`, but actually, here we:
1. Get all the data from saved meta data dictionaries in a for loop directly, not like the iterations append / extend of Cumulative.
2. It `concat` the result instead of `stack` the result in the `sync` of Cumulative:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/metrics/metric.py#L255

So I feel maybe we don't need to use `Cumulative` here? Or could you please help share more details about your suggestion?

Thanks in advance.

Reply:
Sure, I can look into this later, it's not a blocking issue.

Reply:
Thanks, after some expepiments, I have some ideas but need to change the API or args, maybe we can have an online discussion about this topic when you looking into it later.

Thanks in advance.